src/com/owncloud/android/ui/activity/ShareActivity.java

```diff
@@ -116,25 +116,42 @@ protected void onNewIntent(Intent intent) {
         }
     }
 
-    private void doShareWith(String shareeName, boolean isGroup) {
+    private void doShareWith(String shareeName, String dataAuthority) {
+
+        int lastPointIndex = dataAuthority.lastIndexOf(".");
+        String type = dataAuthority.substring(lastPointIndex + 1);
+        ShareType shareType = null;
+        switch (type) {
```

Comment:
The switch could be substituted via a static final hashmap that has been statically prefilled during classloadinig (basically implementing a lookup table) but I leave this idea up for @davivel's vote :smiley_cat: